### PR TITLE
feat: poll recovery state of Safe

### LINKED ISSUE #2752: [Recovery] Proposal data loading/caching
## What is the feature about

Loading and caching of the recovery module(s), proposal(s) and the appropriate caching/polling. Note: all to be done on-chain as per [the proof of concept](https://github.com/safe-global/safe-wallet-web/pull/2537).

## The list of requirements

- [x] Parse modules to determine if Delay Modifier is enabled
- [x] Create Redux slice for new state
- [x] Load module details: recoverers, nonce
- [x] Load proposals after nonce

## Links

- [Feature spec.](https://www.notion.so/safe-global/Recovery-feature-spec-d873c33a27c24cfabc91c85dbcf3ea35)
- [Proof of concept](https://github.com/safe-global/safe-wallet-web/pull/2537/files#diff-484b2028bddb57e28f764fd4f808c4e4cf8b7581a6c169876b02e09abe55a22a)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2752

## How this PR fixes it

This adds a new `recovery` slice to Redux. When a Safe is opened, and subsequently every ten minutes, the modules of it are parsed and for every Delay Modifier, details loaded.

```ts
type RecoveryQueueItem = TransactionAddedEvent & {
  timestamp: number
  validFrom: BigNumber
  expiresAt: BigNumber | null
}

type RecoveryState = Array<{
  address: string
  modules: Array<string>
  txExpiration: BigNumber
  txCooldown: BigNumber
  txNonce: BigNumber
  queueNonce: BigNumber
  queue: Array<QueuedTransactionAdded>
}>
```

## How to test it

Open a Safe with a Delay Modifier enabled, e.g. `gor:0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b` and observe the state load in Redux.

## Screenshots

image

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/hooks/loadables/useLoadRecovery.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          : validFrom.add(txExpiration)
+
+        return {
+          ...event,
+          timestamp: txBlock.timestamp,
+          validFrom,
+          expiresAt,
+        }
+      }),
+  )
```
Comment (katspaugh) [inline src/hooks/loadables/useLoadRecovery.ts:50]: I would extract this into its own function.

**Review by katspaugh — COMMENTED** [review]
Some unit tests would be good.

File: src/hooks/loadables/useLoadRecovery.ts (hunk 1 above)
Comment (iamacook) [reply src/hooks/loadables/useLoadRecovery.ts]: I've extracted it and added test coverage for it (and all other loading related logic) in 181b21d9cfe796e6960f6ba52f531759e34f7184.

File: cypress/e2e/safe-apps/tx-builder.spec.cy.js [hunk 2]
```diff
@@ -146,8 +146,8 @@ describe('Tx-builder Safe App tests', { defaultCommandTimeout: 20000 }, () => {
       getBody().findByText(safeapps.createBatchStr).click()
       getBody().findByText(safeapps.sendBatchStr).click()
     })
-    cy.get('p').contains('1').should('be.visible')
-    cy.get('p').contains('2').should('be.visible')
+    cy.get('p').contains('1').should('exist')
+    cy.get('p').contains('2').should('exist')
```
Comment (iamacook) [inline cypress/e2e/safe-apps/tx-builder.spec.cy.js:150]: This was failing as in https://github.com/safe-global/safe-wallet-web/pull/2768#discussion_r1386237622.

File: src/hooks/loadables/useLoadRecovery.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  return {
+    address: delayModifier.address,
+    modules,
+    txExpiration,
+    txCooldown,
+    txNonce,
+    queueNonce,
+    queue,
+  }
+}
```
Comment (katspaugh) [inline src/hooks/loadables/useLoadRecovery.ts:77]: One might ask at this point why they are in the hook and not in the services/recovery, but up to you.

**Review by katspaugh — APPROVED** [review]

File: src/hooks/loadables/useLoadRecovery.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+export const _getRecoveryState = async (delayModifier: Delay): Promise<RecoveryState[number]> => {
+  const transactionAddedFilter = delayModifier.filters.TransactionAdded()
+
+  const [[modules], txExpiration, txCooldown, txNonce, queueNonce, transactionsAdded] = await Promise.all([
+    delayModifier.getModulesPaginated(SENTINEL_ADDRESS, MAX_PAGE_SIZE),
+    delayModifier.txExpiration(),
+    delayModifier.txCooldown(),
+    delayModifier.txNonce(),
+    delayModifier.queueNonce(),
+    delayModifier.queryFilter(transactionAddedFilter),
```
Comment (schmanu) [inline src/hooks/loadables/useLoadRecovery.ts:57]: I think on some networks the size of logs that can be queried is limited. 
And in general it is adviced to query the on-chain data in chunks (see [here](https://docs.chainstack.com/docs/understanding-eth-getlogs-limitations))

A good first step could be to at least limit the starting block to the block in which the Safe was deployed or the module was enabled. And for later polls we could remember the last checked block and start polling from there.

Comment (iamacook) [reply src/hooks/loadables/useLoadRecovery.ts]: I've created a separate [issue](https://github.com/safe-global/safe-wallet-web/issues/2792) to focus on this specifically. However, I adjusted the query logic to only request from the Safe creation block in f70fff886761a640fe434d8af3f023880fe16690. What do you think?

File: src/hooks/loadables/useLoadRecovery.ts (hunk 3 above)
Comment (iamacook) [reply src/hooks/loadables/useLoadRecovery.ts]: I moved the parsing logic to the `@/services/recovery` folder in d646122711012b911a69c58d796815828709d7e1.

**Review by schmanu — APPROVED** [review]
LGTM! We can tackle the optimization in a separate issue :)

**Pull request merged**